### PR TITLE
Add GCC compile for FM33A0XX

### PR DESCRIPTION
**Pull request by Issacpeng**

### CODE REVIEW

File: targets/Cloud_FM33A0XX/RTE/Device/FM33A04XX/system_FM33A0XX.c [hunk 1]
```diff
@@ -0,0 +1,92 @@
+/**************************************************************************//**
+ * @file     system_<Device>.c
```
Comment (w00164089) [inline targets/Cloud_FM33A0XX/RTE/Device/FM33A04XX/system_FM33A0XX.c]: The file name is  system_FM33A0XX.c

File: targets/Cloud_FM33A0XX/RTE/Device/FM33A04XX/system_FM33A0XX.c [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ * @brief    CMSIS Cortex-M# Device Peripheral Access Layer Source File for
+ *           Device <Device>
+ * @version  V3.10
+ * @date     23. November 2012
+ *
+ * @note
+ *
+ ******************************************************************************/
+/* Copyright (c) 2012 ARM LIMITED
```
Comment (w00164089) [inline targets/Cloud_FM33A0XX/RTE/Device/FM33A04XX/system_FM33A0XX.c]: Notice the license.

File: targets/Cloud_FM33A0XX/GCC/Makefile [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+ARCH_SRC =  \
+        ${wildcard $(TOP_DIR)/arch/arm/arm-m/src/*.c}
+        C_SOURCES += $(ARCH_SRC)
+
+EXAMPLES_SRC =  \
+        ${wildcard $(TOP_DIR)/examples/api/*.c}
+        C_SOURCES += $(EXAMPLES_SRC)
+
+HAL_DRIVER_SRC =  \
```
Comment (w00164089) [inline targets/Cloud_FM33A0XX/GCC/Makefile]: The driver directory is supported by 3rd. May add a license claim?

**Comment by xuliqun25** [comment]
targets/Cloud_FM33A0XX/RTE/RTE_Components.h
这个文件是keil生成的，应该不需要合入。

**Pull request merged**